scripts/sync-to-aws-homebrew-tap

```diff
@@ -0,0 +1,233 @@
+DRY_RUN=0
+
+GH_CLI_VERSION="0.10.1"
+GH_CLI_CONFIG_PATH="${HOME}/.config/gh/config.yml"
+KERNEL=$(uname -s | tr '[:upper:]' '[:lower:]')
+OS="${KERNEL}"
+if [[ "${KERNEL}" == "darwin" ]]; then 
+  OS="macOS"
+fi
+
+VERSION_REGEX="^v[0-9]+\.[0-9]+\.[0-9]+\$"
+VERSION=$(make -s -f "${SCRIPTPATH}/../Makefile" version)
+
+USAGE=$(cat << EOM
+  Usage: sync-to-aws-homebrew-tap  -r <repo> -b <binary-basename> -p <platform_pairs>
+  Syncs tar.gz\'d binaries to the aws/homebrew-tap
+
+  Example: sync-to-aws-homebrew-tap -r "aws/amazon-ec2-instance-selector"
+          Required:
+            -b          Binary basename (i.e. -b "ec2-instance-selector")
+
+          Optional:
+            -r          Github repo to sync to in the form of "org/name"  (i.e. -r "aws/amazon-ec2-instance-selector") [DEFAULT: output of \`make repo-full-name\`]
+            -v          VERSION: The application version of the docker image [DEFAULT: output of \`make version\`]
+            -p          Platform pair list (os/architecture) [DEFAULT: linux/amd64]
+            -d          Dry-Run will do all steps except pushing to git and opening the sync PR
+EOM
+)
+
+# Process our input arguments
+while getopts "p:b:r:v:d" opt; do
+  case ${opt} in
+    r ) # Github repo
+        REPO="$OPTARG"
+      ;;
+    b ) # binary basename
+        BINARY_BASE="$OPTARG"
+      ;;
+    p ) # Supported Platforms
+        IFS=',' read -ra PLATFORMS <<< "$OPTARG"
+      ;;
+    v ) # App Version
+        VERSION="$OPTARG"
+      ;;
+    d ) # Dry Run
+        DRY_RUN=1
+      ;;
+    \? )
+        echo "$USAGE" 1>&2
+        exit
+      ;;
+  esac
+done
+
+if [[ -z "${BINARY_BASE}" ]]; then 
+  echo "Binary Basename (-b) must be specified"
+  exit 3
+fi
+
+if [[ ! "${VERSION}" =~ $VERSION_REGEX ]]; then
+  echo "🙈 Not on a current release, so not syncing with tap $TAP_REPO"
+  exit 3
+fi
+
+if [[ -z "${REPO}" ]]; then 
+  echo "Repo (-r) must be specified if no \"make repo-full-name\" target exists"
+fi
+
+if [[ -z $(command -v gh) ]] || [[ ! $(gh --version) =~ $GH_CLI_VERSION ]]; then
+  mkdir -p ${BUILD_DIR}/gh
+  curl -Lo ${BUILD_DIR}/gh/gh.tar.gz "https://github.com/cli/cli/releases/download/v${GH_CLI_VERSION}/gh_${GH_CLI_VERSION}_${OS}_amd64.tar.gz"
+  tar -C ${BUILD_DIR}/gh -xvf "${BUILD_DIR}/gh/gh.tar.gz"
+  export PATH="${BUILD_DIR}/gh/gh_${GH_CLI_VERSION}_${OS}_amd64/bin:$PATH"
+  if [[ ! $(gh --version) =~ $GH_CLI_VERSION ]]; then
+    echo "❌ Failed install of github cli"
+    exit 4
+  fi
+fi
+
+function restore_gh_config() {
+  mv -f "${GH_CLI_CONFIG_PATH}.bkup" "${GH_CLI_CONFIG_PATH}" || :
+}
+
+if [[ -n $(env | grep GITHUB_TOKEN) ]] && [[ -n "${GITHUB_TOKEN}" ]]; then
+  trap restore_gh_config EXIT INT TERM ERR
+  mkdir -p "${HOME}/.config/gh"
+  cp -f "${GH_CLI_CONFIG_PATH}" "${GH_CLI_CONFIG_PATH}.bkup" || :
+  cat << EOF > "${GH_CLI_CONFIG_PATH}"
+hosts:
+    github.com:
+        oauth_token: ${GITHUB_TOKEN}
+        user: ${GITHUB_USERNAME}
+EOF
+fi
```

Comment:
should the script fail if GITHUB_TOKEN is not set?

Reply:
hmm so I would've said a GITHUB_TOKEN is not necessarily needed if the gh config had a valid OAUTH token, but I violated my own rule further down.  So, as of now, yes, it will require a GITHUB_TOKEN, but I can refactor in another PR to use the gh CLI to do the github api calls and then it will use whatever is in the gh config which could be a personal access token (GITHUB_TOKEN) or an OAUTH token received from a web auth.